### DISCUSSION TITLE
REQUESTS requirement in Python

### DISCUSSION THREAD

**Discussion by anonyada** [GerryDazoo/Slinger#43] [General]
Fabulous piece of work!  I'd like to better understand the requirements of the "dynamic key" build and the "keys.dat" file you create on the initial server start if they are not present.  Are you receiving credentials, serial numbers, ID's etc and then returning the keys.dat file via your personal WAN cluster?  Also....could you elaborate on: "sbfinder.dazoo.net,  54321" in the python source?  Are you building a library of open ports and IP's where users of this code python have established a connection with their SB's?

Inquiring minds.....

Thanks!

Anon--

P.S.  I am currently utilizing a refined PERL script, RPi4B, FIFO, and streamed to FFMPEG on RPi-->VLC.  Would love to see the next version that does not "phone home"!

**Comment by GerryDazoo** [comment]
I’m updating the readme as we speak. The code no longer needs to download the keys from the web which was because I had a bit of perl code that needed to run and I didn’t want people to have to install it on their PCs. 

 

The sbfinder stuff is my attempt to help people who want to remote access their server but don’t have a static IP. Something the Sling code supported. The python code registers their IP address and  server port number and then they can connect to the server from outside there local network by going to sbfinder and supplying their finderid ( courteously supplied by Mr. Sling). Note this  is for finding the python server code not the slingboxes.

So if you don’t want this feature you just don’t configure the finderid in the config.ini file for the slingbox and nothing leaves the local network.

 

From: anonyada ***@***.***> 
Sent: Friday, September 09, 2022 3:24 PM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] REQUESTS requirement in Python (Discussion #43)

 

Fabulous piece of work! I'd like to better understand the requirements of the "dynamic key" build and the "keys.dat" file you create on the initial server start if they are not present. Are you receiving credentials, serial numbers, ID's etc and then returning the keys.dat file via your personal WAN cluster? Also....could you elaborate on: "sbfinder.dazoo.net, 54321" in the python source? Are you building a library of open ports and IP's where users of this code python have established a connection with their SB's?

Inquiring minds.....

Thanks!

Anon--

P.S. I am currently utilizing a refined PERL script, RPi4B, FIFO, and streamed to FFMPEG on RPi-->VLC. Would love to see the next version that does not "phone home"!

**Reply by anonyada** [reply]
Nice!  Appreciate the quick response.

So there is a newer version of the Python source that incorporates these changes?

--Anon

**Reply by anonyada** [reply]
Also....if you'd like my research notes on my PERL adaptation on the RPi4B, I'd be happy to share.  Just let me know

**Reply by GerryDazoo** [reply]
The latest version has these changes. The documentation was out of date. Unfortunately the requests module was still in the imports list in v3.07 but it is not used. You can delete that line if you like. V3.08 is on the way.
